src/2sv/key/Insert.vue

```diff
@@ -21,26 +27,75 @@
 
       <v-spacer></v-spacer>
       
-      <v-btn v-if="isSupported" to="/2sv/usb-security-key/touch" color="primary" outlined>
+      <v-btn v-if="isSupported && showLabelInput" @click="onContinue" color="primary" outlined>
+        {{ $vuetify.lang.t('$vuetify.global.button.continue') }}
+      </v-btn>
+
+      <v-btn v-else-if="isSupported  && !showLabelInput" @click="onOk" color="primary" outlined>
         {{ $vuetify.lang.t('$vuetify.2sv.key.insert.button.ok') }}
       </v-btn>
       <v-btn v-else to="/2sv/printable-backup-codes/intro" color="primary" outlined> 	
         {{ $vuetify.lang.t('$vuetify.global.button.skip') }}	
       </v-btn>
     </ButtonBar>
+
+    <v-snackbar v-model="snackbarIsOpen">
+      {{ snackBarMessage }}
+      
+      <template v-slot:action="{ attrs }">
+        <v-btn
+          color="warning"
+          text
+          v-bind="attrs"
+          @click="snackbarIsOpen = false"
+        >
+           {{ $vuetify.lang.t('$vuetify.global.button.close') }}
+        </v-btn>
+      </template>
+    </v-snackbar>
   </ProfileWizard>
 </template>
 
 <script>
 import ProfileWizard from '@/profile/ProfileWizard'
 import { browserSupportsWebauthn } from '@simplewebauthn/browser';
+import { newKeyName, mfa } from '@/global/mfa'
 
 export default {
   components: {
     ProfileWizard,
   },
   data: () => ({	
     isSupported: browserSupportsWebauthn(),
+    showLabelInput: false,
+    input: '',
+    newKeyName,
+    snackbarIsOpen: false,
+    snackBarMessage: '',
   }),
+  methods: {
+    onOk: function() {
+      this.showLabelInput = true
+    },
+    onKeyup: function(event) {
+      if(event.key == 'Enter') {
+        this.onContinue()
+      }
+    },
+    onContinue: function() {
+      if(!this.input.trim()) {
+        this.snackBarMessage = this.$vuetify.lang.t('$vuetify.2sv.key.insert.label')
+        this.snackbarIsOpen = true
+        return
+      } else if (mfa.keys.data?.find(key => key.label == this.input.trim())) {
```

Comment:
While this looks like it works, would you mind extracting the logic in this if-condition to a separate function, for readability? Something like this...
```js
} else if (isDuplicateLabel(mfa.keys, attemptedKeyLabel)) {
```
... though that assumes extracting/assigning `this.input.trim()` to a new `attemptedKeyLabel` value near the top of onContinue.

Feel free to adjust the specifics however you want.

Reply:
no worries. I had to call `this.isDuplicateLabel` which makes Vue start to feel a little old compared to Svelte.